.github/actions/setup_nox/action.yml

```diff
@@ -0,0 +1,33 @@
+name: Setup Nox
+description: 'Prepare and cache nox, install poetry.'
+
+inputs:
+  cache_number:
+    description: "A manually specified cache number. Useful for triggering new caches."
+    required: True
+    default: 0
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/setup-python@v3
+      with:
+        python-version: "3.7"
+    - uses: actions/setup-python@v3
+      with:
+        python-version: "3.8"
+    - uses: actions/setup-python@v3
+      with:
+        python-version: "3.9"
+    - uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: "Install nox"
+      run: pipx install nox
+      shell: bash
+    - name: "Cache Nox"
+      uses: actions/cache@v2
+      with:
+        path: .nox
+        key:
+          nox-${{inputs.cache_number }}-${{ hashFiles('prql-python/noxfile.py') }}-${{ hashFiles('prql-python/requirements.txt') }}
```

Comment:
FYI sometimes people provide actions that embed this logic; e.g. the rust cache one we use does this